cppcache/src/TcrConnection.hpp

```diff
@@ -33,22 +33,28 @@
 #include "util/synchronized_set.hpp"
 
 #define DEFAULT_TIMEOUT_RETRIES 12
-#define PRIMARY_SERVER_TO_CLIENT 101
-#define SECONDARY_SERVER_TO_CLIENT 102
-#define SUCCESSFUL_SERVER_TO_CLIENT 105
-#define UNSUCCESSFUL_SERVER_TO_CLIENT 106
-#define CLIENT_TO_SERVER 100
-#define REPLY_OK 59
-#define REPLY_REFUSED 60
-#define REPLY_INVALID 61
-#define REPLY_SSL_ENABLED 21
-#define REPLY_AUTHENTICATION_REQUIRED 62
-#define REPLY_AUTHENTICATION_FAILED 63
-#define REPLY_DUPLICATE_DURABLE_CLIENT 64
-
-#define SECURITY_CREDENTIALS_NONE 0
-#define SECURITY_CREDENTIALS_NORMAL 1
-#define SECURITY_MULTIUSER_NOTIFICATIONCHANNEL 3
+
+enum class acceptor : ::std::int8_t {
+  CLIENT_TO_SERVER = 100,
+  PRIMARY_SERVER_TO_CLIENT = 101,
+  SECONDARY_SERVER_TO_CLIENT = 102
+};
+enum class acceptance_codes : ::std::int8_t {
+  REPLY_SSL_ENABLED = 21,
+  REPLY_OK = 59,
+  REPLY_REFUSED = 60,
+  REPLY_INVALID = 61,
+  REPLY_AUTHENTICATION_REQUIRED = 62,
+  REPLY_AUTHENTICATION_FAILED = 63,
+  REPLY_DUPLICATE_DURABLE_CLIENT = 64,
+  SUCCESSFUL_SERVER_TO_CLIENT = 105,
+  UNSUCCESSFUL_SERVER_TO_CLIENT = 106
+};
+enum class security : ::std::uint8_t {
+  SECURITY_CREDENTIALS_NONE = 0,
+  SECURITY_CREDENTIALS_NORMAL = 1,
+  SECURITY_MULTIUSER_NOTIFICATIONCHANNEL = 3
+};
```

Comment:
Really love the separation of concerns here, it's much easier now to tell in the implementation that the switch statements are correct, for instance.  Quick question - do any of these values need to be in the header?  They don't seem to be referenced anywhere outside of TcrConnection code...

Reply:
Ugh, build break - looks like REPLY_SSL_ENABLED is also used in ThinClientLocatorHelper :|

Reply:
Good call, I moved what I could to the source file.